### ISSUE TITLE
[AKS] Restarting AKS will trigger a circular dependency on `admission-webhook` 

### ISSUE THREAD

**Issue by VariableDeclared** [canonical/bundle-kubeflow#420]
Hi,

When deploying Charmed Kubeflow on AKS, kubeflow will prevent the cluster from restarting sucessfully, this is due to the `admission-webhook` pod getting stuck in it's `juju-pod-init` container.

Removing the mutatingwebhookconfiguration using kubectl resolves the situation and allows the cluster to heal.

[Steps to reproduce]

```
az aks create -g kubeflow-repro -n aks-repro-2 --kubernetes-version 1.21.7 --node-count 2 --node-vm-size Standard_B4ms --dns-name-prefix kf-sndbox -l westeurope --generate-ssh-keys
juju add-k8s --aks --resource-group kubeflow-repro --cluster-name aks-repro-2 aks-repro-br --storage default
juju bootstrap aks-repro-br
juju add-model kubeflow 
juju deploy cs:kubeflow-lite

wait for the model to settle.
#### Reboot the cluster
az aks stop -n aks-repro-2 -g kubeflow-repro
# Wait for stop, then start the cluster,

az aks start -n aks-repro-2 -g kubeflow-repro
Observe the admission webhook gets stuck with:

kubectl get pods -A
```

Expected output:

```
Every 2.0s: kubectl get pods -A --context aks-repro-2                                  

Every 2.0s: kubectl get pods -A --context aks-repro-2                                            

NAMESPACE     NAME                                  READY   STATUS              RESTARTS   AGE
kube-system   azure-ip-masq-agent-mg7jx             1/1     Running             0          32s
kube-system   azure-ip-masq-agent-p4v2n             1/1     Running             0          22s
kube-system   coredns-845757d86-n7ctv               1/1     Running             0          8m42s
kube-system   coredns-845757d86-zfx9n               1/1     Running             0          8m42s
kube-system   coredns-autoscaler-5f85dc856b-pjdwq   1/1     Running             0          8m12s
kube-system   csi-azuredisk-node-k7mwr              3/3     Running             0          32s
kube-system   csi-azuredisk-node-xs4kl              3/3     Running             0          22s
kube-system   csi-azurefile-node-shsdf              3/3     Running             0          22s
kube-system   csi-azurefile-node-zggsb              3/3     Running             0          32s
kube-system   kube-proxy-8hnfv                      1/1     Running             0          22s
kube-system   kube-proxy-llw27                      1/1     Running             0          32s
kube-system   metrics-server-6bc97b47f7-rns96       0/1     Running             0          9m12s
kube-system   tunnelfront-f86fb4dbc-hmc8z           1/1     Running             0          8m12s
kubeflow      admission-webhook-7cd7d7796f-gg8tz    0/1     Init:0/1            0          8m12s
kubeflow      admission-webhook-operator-0          0/1     ContainerCreating   0          9m12s

```

[Workaround]

```
kubectl get mutatingwebhookconfigurations kubeflow-admission-webhook  -o yaml > kubeflow-webhook-config.yaml
kubectl delete mutatingwebhookconfigurations kubeflow-admission-webhook
# ..... wait for the cluster to heal. Waiting for admission-webhook to get to Running state.
# delete the uid and version elements of the YAML
kubectl apply -f kubeflow-webhook-config.yaml
```

Thanks,
Peter

**Comment by VariableDeclared** [comment]
hey @ca-scribner yes, here's the logs from the initContainer:

```
➜  aks kubectl logs -n kubeflow admission-webhook-575c9647c6-rpf5s -c juju-pod-init
2022-01-13 11:46:08 INFO juju.cmd supercommand.go:56 running jujud [2.9.22 0 07164c4b2c6a9f5b576b1c151bd85dd3a699be16 gc go1.14.15]
2022-01-13 11:46:08 DEBUG juju.cmd supercommand.go:57   args: []string{"/var/lib/juju/tools/jujud", "caas-unit-init", "--debug", "--wait"}
➜
```

**Comment by Barteus** [comment]
I have tried to apply this workaround twice. Once it worked fine, but the second time the kubeflow-webhook-config.yaml was not applied correctly because the configuration of it changed meanwhile during kubeflow restart.

**Comment by DomFleischmann** [comment]
I have been able to reproduce this issue and we will look into a possible fix.

**Comment by DnPlas** [comment]
Hi @VariableDeclared, thanks for reporting this. I was able to reproduce the issue on AKS, just as you described. Apparently this is an issue with juju, in fact there have been multiple reports similar to this one. I have filed a [bug](https://bugs.launchpad.net/juju/+bug/1973383) (as the juju team requested) so we can find out a reliable workaround or a solution.
In the meantime, we can use your proposed workaround.
